### PR TITLE
Add new endpoint for deleting multiple timereports at once

### PR DESCRIPTION
**Pull request by neurogoo**
TimereportDelete type needed extra data type to get ToJson and FromJson to work. Why is that and can it be fixed some other way? I think there was similar problem previously, but can't remember where

### CODE REVIEW

File: hours-api/src/Futurice/App/HoursApi/API.hs [hunk 1]
```diff
@@ -29,6 +29,7 @@ type FutuhoursV1API =
     :<|> "entry" :> SSOUser :> ReqBody '[JSON] EntryUpdate :> Post '[JSON] EntryUpdateResponse
     :<|> "entry" :> SSOUser :> Capture "id" PM.TimereportId :> ReqBody '[JSON] EntryUpdate :> Put '[JSON] EntryUpdateResponse
     :<|> "entry" :> SSOUser :> Capture "id" PM.TimereportId :> Delete '[JSON] EntryUpdateResponse
+    :<|> "delete-timereports" :> SSOUser :> ReqBody '[JSON] TimereportDelete :> Post '[JSON] EntryUpdateResponse
```
Comment (phadej) [inline hours-api/src/Futurice/App/HoursApi/API.hs]: Why not simply `[PM.TimereportId]` ? i.e. why wrap in single-field object?

File: hours-api/src/Futurice/App/HoursApi/Logic.hs [hunk 2]
```diff
@@ -94,6 +95,19 @@ entryDeleteEndpoint eid = do
...
+entryDeleteMultipleEndpoint :: H.MonadHours m => TimereportDelete -> m EntryUpdateResponse
+entryDeleteMultipleEndpoint (TimereportDelete eids _) = do
+    let deleteTimereport eid = do
+            tr <- H.timereport eid
+            _ <- H.deleteTimereport eid
+            pure tr
+        days = sort . map (^. H.timereportDay)
+        firstDay = head . days
+        lastDay = last . days
+    trs <- mapM deleteTimereport eids
```
Comment (phadej) [inline hours-api/src/Futurice/App/HoursApi/Logic.hs]:
```haskell
for_ eids $ \eid -> do
    tr <- H.timereport eid
    _ <- H.depeteTimereport eid
    pure tr
EntryUpdateResponse <$> ...
```

---

What if `eids` is empty? Use `NonEmpty`, not list.

File: hours-api/src/Futurice/App/HoursApi/API.hs (hunk 1 above)
Comment (neurogoo) [reply hours-api/src/Futurice/App/HoursApi/API.hs]: It felt that if would be a little clearer, but I can change that to that :)

**Comment by neurogoo** [comment]
Feels somewhat clumsy solution. There must be a better way to write it?

**Pull request merged**